### PR TITLE
fix: [CDS-45259]: github packages registry support for serviceV2

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ServiceV2ArtifactsSelection.tsx [hunk 1]
```diff
@@ -132,6 +132,13 @@ export default function ServiceV2ArtifactsSelection({
     }
     if (
       deploymentType === ServiceDeploymentType.Kubernetes &&
+      GITHUB_PACKAGES &&
+      !allowedArtifactTypes[deploymentType]?.includes(ENABLED_ARTIFACT_TYPES.GithubPackageRegistry)
+    ) {
+      allowedArtifactTypes[deploymentType].push(ENABLED_ARTIFACT_TYPES.GithubPackageRegistry)
+    }
+    if (
+      ['Kubernetes', 'CustomDeployment'].includes(deploymentType) &&
```
Comment (Sagar-Rajput7) [inline src/modules/70-pipeline/components/ArtifactsSelection/ServiceV2ArtifactsSelection.tsx:141]: restricting to this is expected right?

Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ServiceV2ArtifactsSelection.tsx:141]: yes

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**